### PR TITLE
Guard check that event_base_new initializes correctly

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos sonanopo@microsoft.com

Add a release assert to the libevent `event_base_new()` to make sure that the initialization is correct and make envoy fail fast if it fails.

Additional Description: 

This scenario crashes with an Access Violation at later stages. With this change we just make sure that we crash closer to where we actually failed. 

Risk Level: low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
cc the usual windows crew @wrowe @sunjayBhatia @nigriMSFT

File: source/common/event/libevent_scheduler.cc [hunk 1]
```diff
@@ -15,7 +15,11 @@ void recordTimeval(Stats::Histogram& histogram, const timeval& tv) {
 }
 } // namespace
 
-LibeventScheduler::LibeventScheduler() : libevent_(event_base_new()) {
+LibeventScheduler::LibeventScheduler() {
+  auto eventBase = event_base_new();
```
Comment (jmarantz) [inline source/common/event/libevent_scheduler.cc:19]: event_base per style guide.

Comment (davinci26) [reply source/common/event/libevent_scheduler.cc]: Thanks, it should be fixed in 0c2f5f467686c2ee1afed552a24f08af1f8cad2b

File: source/common/event/libevent_scheduler.cc [hunk 2]
```diff
@@ -15,7 +15,11 @@ void recordTimeval(Stats::Histogram& histogram, const timeval& tv) {
 }
 } // namespace
 
-LibeventScheduler::LibeventScheduler() : libevent_(event_base_new()) {
+LibeventScheduler::LibeventScheduler() {
+  event_base* eventBase = event_base_new();
```
[the remark this answers is not in the document]
Comment (davinci26) [reply source/common/event/libevent_scheduler.cc]: thanks totally missed this one

**Pull request merged**